components/raftstore-v2/src/operation/ready/async_writer.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    collections::VecDeque,
+    sync::{atomic::AtomicUsize, Arc},
+};
+
+use crossbeam::channel::Sender;
+use engine_traits::{KvEngine, RaftEngine};
+use kvproto::raft_serverpb::RaftMessage;
+use raftstore::store::{
+    local_metrics::RaftMetrics, Config, PersistedNotifier, WriteMsg, WriteRouter,
+    WriteRouterContext, WriteSenders, WriteTask,
+};
+use slog::{warn, Logger};
+
+use crate::{
+    batch::{StoreContext, StoreRouter},
+    PeerMsg,
+};
+
+#[derive(Debug)]
+struct UnpersistedReady {
+    /// Number of ready.
+    number: u64,
+    /// Max number of following ready whose data to be persisted is empty.
+    max_empty_number: u64,
+    raft_msgs: Vec<Vec<RaftMessage>>,
+}
+
+/// A writer that handles asynchronous writes.
+pub struct AsyncWriter<EK: KvEngine, ER: RaftEngine> {
```

Comment:
I cannot find its counterpart of v1. Which structure in v1 has similar function?

Reply:
There is no such struct in v1. It's a reorganized of code. In v1, async write is highly clustered with the ready processing.